### ISSUE TITLE
no-sync rule matches properties

### ISSUE THREAD

**Issue by pbialek** [eslint/eslint#6880]
**What version of ESLint are you using?**
ESLint v3.2.2.

**ESLint configuration:**
```
{
    "rules": {
        "no-sync": 2
    }
}
```

**Source code:**
```
var foo = {
    barSync: 13
}
console.log(foo.barSync);
```

**What did you expect to happen?**
`barSync` is a property of `foo` object. I suppose `no-sync` shouldn't check it.

**What actually happened? Please include the actual, raw output from ESLint.**
```
  4:13  error  Unexpected sync method: 'barSync'  no-sync

✖ 1 problem (1 error, 0 warnings)
```

**Comment by mysticatea** [comment]
Thank you for this issue.

I think the rule should warn properties because the purpose of the rule is to prevent a use of `fs.readFileSync`-like methods.

**Comment by pbialek** [comment]
Right, I fully agree, that `fs.readFileSync`-like methods should be checked. To be honest I've applied `no-sync` ESLint rule to forbid using I/O blocking functions from `fs`.
But as you can see, `barSync` is not a method.

What about checking if it is a callee or if next token is `(` similar to [func-call-spacing](https://github.com/eslint/eslint/blob/master/lib/rules/func-call-spacing.js) rule?

**Comment by platinumazure** [comment]
@pbialek Then there would be no way to catch uses like this:

```js
var fileNames = getFileNames();
var contents = fileNames.map(fs.readFileSync);
```

I think a better approach here would be to enhance the rule to allow ignoring certain patterns.

Don't forget in the meanwhile, you can use inline configuration comments to disable the rule for those cases.

**Comment by pbialek** [comment]
Ok. You are right. I forgot about passing function as an argument.
